inc/packages/namespace.php

```diff
@@ -524,9 +533,6 @@ function delete_package( string $slug, bool $for_network = false ) {
 		);
 	}
 
-	// Update the index settings on successful deletion.
-	do_settings_update( $for_network, strpos( $slug, 'user-dictionary' ) !== false );
```

Comment:
Do you know any more about what the fatal is or why it happens? How are indexes updated if a package is deleted now?

Reply:
The fatal was happening because [`update_index_settings`](https://github.com/humanmade/altis-enhanced-search/blob/6eab34da0c9ebcc176dc70fa2e3b07a52e975708/inc/packages/namespace.php#L606) called from [`do_settings_update`](https://github.com/humanmade/altis-enhanced-search/blob/6eab34da0c9ebcc176dc70fa2e3b07a52e975708/inc/packages/namespace.php#L589) expects `settings` to not be null, which it was always being passed as null, unsure really why is that happening tbh, didn't quite get the logic in [`elasticpress_mapping`](https://github.com/humanmade/altis-enhanced-search/blob/6eab34da0c9ebcc176dc70fa2e3b07a52e975708/inc/namespace.php#L843) that returns early, but it certainly fatal'd all the time.

And re indexes being updated, that's happening in [`handle_form`](https://github.com/humanmade/altis-enhanced-search/blob/6eab34da0c9ebcc176dc70fa2e3b07a52e975708/inc/packages/namespace.php#L398) where `delete_package` is called (only place I could see it being used). `create_package` does not directly update indexes as well, so seemed like `delete_package` should also be consistent around that, unless I missed something that isn't obvious. IMHO, delete package should not have more side-effects than just deleting the package, consumers should update indexes separately (specially if any bulk operation is happening, otherwise multiple settings update calls can happen).

Reply:
Aha yeah, it's covered by this hook: `add_action( 'altis.search.updated_packages', __NAMESPACE__ . '\\on_updated_packages', 10, 2 );`. Maybe it was something I forgot to remove.

In the cloud module however, we remove that hook in favour of waiting until the packages have been created and associated with the AWS ES domain.

I think we need to take another look at that. Backwards compatibility is useful but there's no need to wait for the package association steps if we're adding the settings inline.